datafusion/src/execution/context.rs

```diff
@@ -896,12 +895,13 @@ impl Default for ExecutionConfig {
             target_partitions: num_cpus::get(),
             batch_size: 8192,
             optimizers: vec![
-                Arc::new(ConstantFolding::new()),
+                // Simplify expressions first to maximize the chance
+                // of applying other optimizations
+                Arc::new(SimplifyExpressions::new()),
```

Comment:
This change has the additional benefit of allowing even more filters to be pushed down (e.g. `column * 1 < 5` which is simplified by the code in `simpify` to `column < 5` can now also be pushed down